src/librbd/image/RefreshParentRequest.cc

```diff
@@ -165,11 +165,17 @@ void RefreshParentRequest<I>::send_set_parent_snap() {
     return;
   }
 
+  cls::rbd::SnapshotNamespace snap_namespace;
```

Comment:
Merge this logic w/ the logic on line 159. This implies we need a new helper method ```ImageCtx::get_snap``` which will populate both the snap name and namespace. This avoids weird errors where you drop the lock and the snapshot disappears. Also fixes the case where the potential error isn't currently handled below.